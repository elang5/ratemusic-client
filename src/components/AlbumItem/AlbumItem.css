@media only screen and (max-width: 400px) {
  .album-img {
    width: 80%;
  }
}

.album-name {
  font-size: 20px;
}