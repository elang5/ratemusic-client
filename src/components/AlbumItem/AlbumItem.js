import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './AlbumItem.css'

export class AlbumItem extends Component {
  render() {
    const { album } = this.props
    return (
      <div className="cell">
        <h3 className="album-rating">{album.rating}</h3>
          <div className="container">
            <Link to={`/albums/${album.id}`} className="album">
              <img src={album.art} alt={album.title}/>
            </Link>
          </div>
        </div>
    )
  }
}

export default AlbumItem
