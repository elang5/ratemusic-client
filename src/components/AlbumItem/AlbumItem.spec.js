import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AlbumItem from './AlbumItem'

describe(`AlbumItem Component`, () => {
  const props = {
    album: '#',
    name: 'test',
    album_id: '12345',
    rating: 8
  }

  it(`renders the AlbumItem component`, () => {
    const wrapper = shallow(<AlbumItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it(`renders the AlbumItem component given props`, () => {
    const wrapper = shallow(<AlbumItem {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})