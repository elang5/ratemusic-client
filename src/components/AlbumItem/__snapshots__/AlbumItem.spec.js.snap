// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumItem Component renders the AlbumItem component 1`] = `
<div
  className="album-item"
>
  <div
    className="album-name-container"
  >
    <h3
      className="album-name"
    />
  </div>
  <div
    className="container"
  >
    <Link
      className="album"
      to="/albums/undefined"
    >
      <img
        className="album-img"
      />
    </Link>
    <AlbumStarRating
      className="review-stars"
    />
  </div>
</div>
`;

exports[`AlbumItem Component renders the AlbumItem component given props 1`] = `
<div
  className="album-item"
>
  <div
    className="album-name-container"
  >
    <h3
      className="album-name"
    >
      test
    </h3>
  </div>
  <div
    className="container"
  >
    <Link
      className="album"
      to="/albums/12345"
    >
      <img
        alt="#"
        className="album-img"
        src="#"
      />
    </Link>
    <AlbumStarRating
      className="review-stars"
      rating={8}
    />
  </div>
</div>
`;
